pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -183,19 +184,19 @@ func (l *podNICImpl) PlugPhase1(vmi *v1.VirtualMachineInstance, iface *v1.Interf
 		return err
 	}
 
-	isExist, err := bindMechanism.loadCachedInterface()
-	if err != nil {
+	if err := bindMechanism.loadCachedInterface(); err != nil {
 		return err
 	}
 
+	doesExist := bindMechanism.hasCachedInterface()
```

Comment:
Is there a use case this will ever return `true` in `Phase1`?
How can we get to `PlugPhase1` with interface in the cache? Caching the interface is done very close to the end of `Phase1`.
If it succeeds, we won't get again to `Phase1`. If the command after `bindMechanism.setCachedInterface` fails, a critical error is returned, so we won't get again to `Phase1`.

Reply:
If virt-handler dies after `bindMechanism.setCachedInterface` is executed but before the VMI [successfully synchs](https://github.com/kubevirt/kubevirt/blob/b7a0b1164be47c34e572977bcc2830d297ba23cf/pkg/virt-handler/vm.go#L2359), I think we reach this situation. 

Am I wrong ? 

I'd also like to point out that the solution I'm proposing here is an attempt at preserving the existent behavior while cleaning up the interface methods; I think that removing this hypothetical dead code deserves a work item on its own.

Reply:
You're right. this is a possible scenario.
But i think it means we have a possible bug here.
If virt-handler crashes before `setCachedVIF` but after `setCachedInterface` we will never invoke `setCachedVIF`. Don't we?
I think the invocation order of `setCachedInterface` and `setCachedVIF` should be switched.

Regarding the fact it is only a refactoring PR. You're right, but you're cleaning this code, so it is a good opportunity to dicuss it.

Reply:
> You're right. this is a possible scenario.
> But i think it means we have a possible bug here.
> If virt-handler crashes before `setCachedVIF` but after `setCachedInterface` we will never invoke `setCachedVIF`. Don't we?

I think you're right. 

> I think the invocation order of `setCachedInterface` and `setCachedVIF` should be switched.

I agree; if the criteria for configuring the network is set by `setCachedInterface`, having that instruction as the last one makes the code more robust (not bullet proof, but better). 

I'll push a separate PR with this change, and we can discuss it / evaluate alternatives over there. Do you agree  ?

> 
> Regarding the fact it is only a refactoring PR. You're right, but you're cleaning this code, so it is a good opportunity to dicuss it.

Don't get me wrong; I'm pretty OK with us uncovering all that is suspicious and makes us frown upon the code in any PR. In fact, we should :)

I just meant we should logkeep separate issues on follow up PRs, or even kubevirt issues depending on their complexity.

Reply:
NP, can be done in a separate PR. Please notice I've added and 'Edit' section to my previous comment.